### PR TITLE
added elytron ldap authentication extension

### PR DESCRIPTION
**Pull request by hartimcwildfly**
see issue #6606 and #5226

### CODE REVIEW

**Comment by hartimcwildfly** [comment]
Please add the execution of elytron ldap integration tests to "Native Tests - Security1".
On my local machine the ITs worked.

**Comment by hartimcwildfly** [comment]
Seems like the step "Quarkus CI / Windows JDK 11 JVM Tests (pull_request)" is stuck.
Running for 2h 24m and I can't see any logs.

**Comment by hartimcwildfly** [comment]
Triggered another run via force push.

**Comment by hartimcwildfly** [comment]
@gsmet this is ready for merge. Don't forget to adjust the CI so the native integration tests are enabled for elytron ldap security.

**Comment by gsmet** [comment]
@hartimcwildfly could you adjust the `ci-actions` file yourself?

Security2 is already a bit overloaded so just add it to Security1 and add 10 mn to the timeout.

**Comment by hartimcwildfly** [comment]
Ok, I will do. Formerly it was saved under ci-templates so initially I could not find where it was defined.

**Comment by hartimcwildfly** [comment]
Done. Let's see if it runs through :)

**Comment by hartimcwildfly** [comment]
Everythings works as expected. But I will check if the classes DelegatingLdapContext and QuarkusDirContextFactory are really needed. (They were added by some of you guys. Not by myself)

**Comment by hartimcwildfly** [comment]
Ok seems to be still needed. When using SimpleDirContextFactoryBuilder from elytron instead of QuarkusDirContextFactory I get some classloading issues in native mode.

--> Ready for merge :)

**Comment by sberyozkin** [comment]
When I looked at it last time, it was all very neat (with only some invisible native issues :-) ), and it looks the same to me now. I'm happy to add my own approval, but please wait for Guillaume to add his approval...

**Review by sberyozkin — APPROVED** [review]

**Comment by hartimcwildfly** [comment]
Just fixed some deprecation warnings :)

File: docs/src/main/asciidoc/security-ldap.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-ldap.adoc:20]:
```suggestion
* Apache Maven {maven-version}
```
(we changed that recently)

File: docs/src/main/asciidoc/security-ldap.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-ldap.adoc:57]:
```suggestion
which is a https://docs.wildfly.org/19/WildFly_Elytron_Security.html#ldap-security-realm[`wildfly-elytron-realm-ldap`] adapter for Quarkus applications.
```

File: docs/src/main/asciidoc/security-ldap.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-ldap.adoc:111]:
```suggestion
We are using `SecurityContext` to get access to the current authenticated Principal and we return the user's name. This information is loaded from the LDAP server.
```

File: docs/src/main/asciidoc/security-ldap.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-ldap.adoc:142]:
```suggestion
----
```

File: docs/src/main/asciidoc/security-ldap.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/security-ldap.adoc:156]:
```suggestion
----
```

File: extensions/elytron-security-ldap/runtime/src/main/java/io/quarkus/elytron/security/ldap/config/IdentityMappingConfig.java
Comment (gsmet) [inline extensions/elytron-security-ldap/runtime/src/main/java/io/quarkus/elytron/security/ldap/config/IdentityMappingConfig.java:15]: Just to be sure, this is mandatory?

**Review by gsmet — CHANGES_REQUESTED** [review]
Nice work. I added a few comments/questions here and there.

BTW, out of curiosity, how did you fix the native image issue?

File: extensions/elytron-security-ldap/runtime/src/main/java/io/quarkus/elytron/security/ldap/config/IdentityMappingConfig.java
Comment (hartimcwildfly) [reply extensions/elytron-security-ldap/runtime/src/main/java/io/quarkus/elytron/security/ldap/config/IdentityMappingConfig.java:15]: It is mandatory, but I can set "uid" to the default value. I think this makes sense.

**Comment by hartimcwildfly** [comment]
Resolved all your proposals and squashed the commits.

I didn't fix the native image by myself. It just worked.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks for your tenacity!

**Comment by hartimcwildfly** [comment]
Thank you, too!

**Pull request merged**
